bin/storm.py

```diff
@@ -1464,18 +1464,32 @@ def print_server_classpath(args):
 def monitor(args):
     exec_storm_class(
         "org.apache.storm.command.Monitor", storm_config_opts=args.storm_config_opts,
-        args=remove_common_options(sys.argv[2:]),
+        main_class_args=remove_common_options(sys.argv[2:]),
         jvmtype="-client",
         extrajars=[USER_CONF_DIR, STORM_BIN_DIR])
 
+def is_not_conflicting_jar_argument(raw_arg):
+    # TODO refactor underlying Topology main classes to avoid this ugly hack
+    conflicting_flags = ["-hdfsConf"]
```

Comment:
Why is it specific to `-hdfsConf`?

Reply:
That's the ugly part - the conflict is that the way the HdfsTopology specifies the argument is non-unix compliant. And there's no consistent way to identify all such possible flags so I'm only filtering this one for now and add more later

Reply:
Any idea why flags starting with `-h` will fail, e.g. `-htt`?
```
storm jar: error: argument -h/--help: ignored explicit argument 'tt'
```

Reply:
Because -htt is equivalent to -h -tt. The correct way to specify it is --htt. The HdfsTopology acceps -htt (which is non-Unix compliant it should be single dashes for single characters or double dashes for multiple characters)

Reply:
I think I finally find a work around: set `allow_abbrev=False`.

Reply:
That doesn't solve the underlying issue - there are single flag arguments that we had previously in the CLI. Setting that will make it backwards incompatible (like the -c)

Reply:
I had an offline discussion with @govind-menon and verified this would work. 
If `allow_abbrev` is true, then `-htt` will be parsed as `-h tt`. And that's why it failed. 

The only problem is `allow_abbrev` is not available in python2.